### PR TITLE
[PoC - do not merge] demangle 'I' as `in`

### PR DESCRIPTION
**Pull request by rainers**
as a companion to https://github.com/dlang/dmd/pull/10526

### CODE REVIEW

**Comment by thewilsonator** [comment]
Whats with the accent in the commit message?

**Comment by rainers** [comment]
> Whats with the accent in the commit message?

Oops, typo corrected.

File: src/core/demangle.d [hunk 1]
```diff
@@ -923,7 +923,6 @@ pure @safe:
             return dst[beg .. len];
         case 'F': case 'U': case 'W': case 'V': case 'R': // TypeFunction
             return parseTypeFunction( name );
-        case 'I': // TypeIdent (I LName)
```
Comment (PetarKirov) [inline src/core/demangle.d]: So `TypeIdent` is not currently being used? Do you know what it was supposed to mean?

Comment (rainers) [reply src/core/demangle.d]: TypeIdentifier is part of the grammar, but it must be resolved to some other entity during semantics. See https://github.com/dlang/dmd/pull/10526/files#diff-ee89b0981cbb7f8ab2412e68a427d52aR100 for other similar "types".

**Comment by rainers** [comment]
Closing as a different implementation of `in = scope const` was merged.

**Pull request closed without merging**